donations/views.py

```diff
@@ -77,43 +77,51 @@ def _save_donation(encoded_data, email, amount, currency, transaction_id, source
 @csrf_exempt
 def donation_complete_stripe(request):
     """
-    This view receives a stripe token of a validated credit card and requests for the actual donation.
-    If the donation is successfully done it stores it and send the email to the user.
+    This view is called from Stripe when a new donation is completed, here we create and
+    store the donation in the db.
     """
-    donation_received = False
-    donation_error = False
-    if request.method == 'POST':
-        form = DonateForm(request.POST, user=request.user)
+    if "HTTP_STRIPE_SIGNATURE" in request.META:
+        payload = request.body
+        sig_header = request.META['HTTP_STRIPE_SIGNATURE']
+        endpoint_secret = settings.STRIPE_WEBHOOK_SECRET
+        stripe.api_key = settings.STRIPE_PRIVATE_KEY
+        event = None
 
-        token = request.POST.get('stripeToken', False)
-        email = request.POST.get('stripeEmail', False)
+        try:
+            event = stripe.Webhook.construct_event(
+              payload, sig_header, endpoint_secret
+            )
+        except ValueError as e:
+            # Invalid payload
+            return HttpResponse(status=400)
+        except stripe.error.SignatureVerificationError as e:
+            # Invalid signature
+            return HttpResponse(status=400)
+
+        # Handle the checkout.session.completed event
+        if event['type'] == 'checkout.session.completed':
+            session = event['data']['object']
+
+            # Fulfill the purchase...
+            amount = int(session['display_items'][0]['amount'])/100
```

Comment:
if I understand correctly this division will truncate in python2. should use `100.0`  for example

Reply:
Stripe handles the amount as integer (multiplied by 100), that is why we divide it for storing it with decimals. 

You're right, this should 100.0 for python 2. Unfortunately, this was not captured by the test cases because of the amounts used.

Reply:
this has not been fixed rigth? will you update the tests as well?

Reply:
Yes, I just submitted the changes.